src/sentry/api/endpoints/project_rule_details.py

```diff
@@ -135,10 +133,20 @@ def get(self, request: Request, project, rule) -> Response:
         # Prepare Rule Actions that are SentryApp components using the meta fields
         for action in serialized_rule.get("actions", []):
             if action.get("_sentry_app_installation") and action.get("_sentry_app_component"):
-                installation = SentryAppInstallation(**action.get("_sentry_app_installation", {}))
```

Comment:
Constructing this ORM object in the Region silo was the source of the problem as `installation.sentry_app` can't work in region silos.